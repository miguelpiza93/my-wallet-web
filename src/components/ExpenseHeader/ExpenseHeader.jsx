import { useNavigate } from "react-router-dom";
import OveralBox from "../OveralBox/OveralBox";

const ExpenseHeader = () => {
  const navigate = useNavigate();
  const handleRedirectToAddExpense = () => {
    navigate("/add");
  };
  return (
    <div>
      <OveralBox />
      <button onClick={handleRedirectToAddExpense}>Add Expense</button>
    </div>
  );
};

export default ExpenseHeader;
