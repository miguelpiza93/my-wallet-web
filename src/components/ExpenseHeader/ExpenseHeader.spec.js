// Import necessary modules at the top
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import ExpenseHeader from "./ExpenseHeader";

// Mocking useNavigate
const mockNavigate = jest.fn()
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate,
  }));
  const mockStore = createMockStore();
  const store = mockStore({
    expenses: [],
  });

  const renderComponent = () => {
    return render(
      <Provider store={store}>
        <MemoryRouter>
          <ExpenseHeader />
        </MemoryRouter>
      </Provider>
    );
  };

describe("ExpenseHeader", () => {
  it("renders the OveralBox component", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('navigates to the add expense page when the "Add Expense" button is clicked', () => {
    renderComponent();
    userEvent.click(screen.getByLabelText("Add Expense"));
    expect(mockNavigate).toHaveBeenCalledWith("/add");
  });
});
