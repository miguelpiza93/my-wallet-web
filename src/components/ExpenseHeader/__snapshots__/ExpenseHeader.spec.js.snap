// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseHeader renders the OveralBox component 1`] = `
<div>
  <div>
    <div>
      <label>
        Total Amount: 
      </label>
       
      0
    </div>
    <button
      aria-label="Add Expense"
    >
      Add Expense
    </button>
  </div>
</div>
`;
